Sources/Testing/Parameterization/Test.Case.swift

```diff
@@ -127,6 +127,26 @@ extension Test {
 
     /// The second name of this parameter, if specified.
     public var secondName: String?
+
+    /// Information about the type of this parameter.
+    ///
+    /// The value of this property represents the type of the parameter, but
+    /// arguments passed to this parameter may be of different types. For
+    /// example, an argument may be a subclass or conforming type of the
+    /// declared parameter type.
+    ///
+    /// For information about runtime type of an argument to a parameterized
+    /// test, use ``TypeInfo/init(describingTypeOf:)``, passing the argument
+    /// value obtained by calling ``Test/Case/Argument/value``.
+    @_spi(ForToolsIntegrationOnly)
+    public var typeInfo: TypeInfo
```

Comment:
Hmm… I don't love "TypeInfo" either, but I worry "TypeName" might be overly restrictive if we do end up evolving this further. I could imagine us wanting to include extra details like what _kind_ of type it represents.

Do you have any other ideas that still leave the door open to this type including other information beyond names? "TypeDetails" is an alternative I considered but I don't think it's much better, and "Type" might be confusing

Reply:
I think if we evolve this much further, it runs the risk of looking like a reflection interface, which is not the intent. I'd rather keep it limited in scope for now and avoid any suggestion that we're trying to build reflection here.

Reply:
I've evaluated "TypeName", but I'm finding that it leads to some redundant-sounding usage sites because it contains properties which also end with the word "name", like `x.typeName.qualifiedTypeName`.

I don't have any plans to expand what this type contains, but conceptually I do think it makes sense to describe it as a type which contains information _about_ a type, such as varying representations of a type's name. It's not, itself, a name—it _contains_ names.

Reply:
But sure, I don't feel strongly here. :)